status/api/common.nim

```diff
@@ -57,3 +59,15 @@ proc initUserDb*(self: StatusObject, keyUid, password: string) {.raises:
   [Exception].} =
 
   self.userDbConn = initializeDB(self.dataDir / keyUid & ".db", password)
+
+proc web3*(self: StatusObject, network: string): Web3 {.raises: [Exception]} =
```

Comment:
```suggestion
proc web3*(self: StatusObject, network: string): Web3 {.raises: [Defect, Exception]} =
```

We should indicate `Defect` in the `{.raises.}` pragma.

It would better to not declare `Exception` in the `{.raises.}` pragma, as it somewhat defeats the purpose (even if it makes the compiler happy). What we really want is to declare specific kinds of exceptions, e.g. `KeyError` or maybe error/s that getSettings could raise.

However, there's a big refactor @emizzle is working on (almost done) related to this (in a general sense), so it's something that can be sorted out in a future PR if [`KeyError`](https://nim-lang.org/docs/system.html#KeyError) isn't enough.